### ISSUE TITLE
Text not getting copied in modal in ngx clipboard

### ISSUE THREAD

**Issue by akshayseth** [maxisam/ngx-clipboard#118]
I am using ngx clipboard to copy my value to clipboard which is working fine if I do that in the main page but I want that functionality in my modal which is neither throwing any error nor copying anything.
Angular-5.2.3
ngx-clipboard-9.1.3
Below is the code:

    <span>
    <img src="../../assets/img/copy-icon.png" ngxClipboard  [cbContent]="myvalue" (click)="copyToClipboard()">
    </span>
and my ts file:

    copyToClipboard(){
      console.log("copyToClipboard")
     }

**Comment by adgoncal** [comment]
It works for me:
https://stackblitz.com/edit/angular-u3tydg-wt7h1t?file=app%2Fdialog%2Fdialog.component.html

**Comment by matthewdfleming** [comment]
## TLDR;
Make your own textarea to hold the value to copy and hook ngxClipboard to it.

## Longer Explanation
I had a couple of issues with this plugin that seemed to **happen a lot in Chrome (v64)**. I was getting intermittent successes.. most of the time the 'copy to clipboard' would **fail** to actually **copy** anything, and the 'success' callback would also be called. So it was the worst of both.. nothing in the logs, no copy, **cbOnSuccess called**. The above seemed exacerbated by **large amounts of text**. Safari seemed to work just fine. In Chrome, I noticed some **messages** in the console in the '**verbose**' mode that were timeout kinds of errors. 

Now I don't know the full story, but it seemed like the code in ngx-clipboard that was '**creating a text area**' was showing me (in the debugger) a message like '**Uncaught SyntaxError: Unexpected end of input**'.

I also use a **Bootstrap modal** but I'm not sure if that was really causing much of an issue. In the end, I was able to get everything going when I:
- made my own `<textarea>`
- used the `[ngxClipboard]="textAreaName"` format

## html template
```
<div class="modal fade" id="export-bundle-modal-{{bundleToExport.id}}" tabindex="-1" role="dialog"
     aria-labelledby="modal-content modal-header title">
    <div class="modal-dialog" role="document">
        <div class="modal-content">
            <div class="modal-header">
                <button type="button" class="close" #exportBundleModalClose data-dismiss="modal">&times;</button>
                <span id="title">Export {{bundleToExport.name}}</span>
            </div>
            <div class="modal-body">
                <div class="row">
                    <div class="col-md-12">
                        <textarea class="form-control text-area" readonly="readonly"
                                  rows="10" #bundleJson style="resize: none">{{bundleWithDependencies | json}}</textarea>
                    </div>
                </div>
            </div>
            <div class="modal-footer">
                <button class="btn btn-success" type="submit" data-dismiss="modal"
                        [disabled]="!taxonomiesLoaded || !componentTypesLoaded || !configurationsLoaded"
                        [ngxClipboard]="bundleJson" (cbOnSuccess)="copied()">
                    <small *ngIf="!(taxonomiesLoaded && componentTypesLoaded && configurationsLoaded)">
                        LOADING...
                    </small>
                    <small *ngIf="taxonomiesLoaded && componentTypesLoaded && configurationsLoaded">
                        COPY TO CLIPBOARD
                    </small>
                </button>
            </div>
        </div>
    </div>
</div>
```
Notice the button and how `bundleJson` is attached to the `<textarea>` via the `#bundleJson` and the 'value' of the textarea is a pipe calculated value. Doing the calculation here (as opposed to when the 'copy to clipboard' was clicked also was necessary.. in essence, make sure the string in the textarea is correct before trying to send it to the clipboard.

We also use the bootstrap.js + jQuery integration instead of a native Angular bootstrap solution (don't ask), so notice that the `data-dismiss="modal"` is also attached to the button. When the button is clicked both the `copied()` function as well as the close of the modal happen.. there didn't seem to be any chaining/timing issues.

**Comment by akshayseth** [comment]
@matthewdfleming thankyou so much for the response.
You are an angel.lmao

**Comment by SehgalDivij** [comment]
@matthewdfleming Hey man, tried the same. Didn't work for me. (\/_\/)

**Comment by akshayseth** [comment]
use this package in textarea only.
